Add test cases to the test suite "changeParams"

HoOngEe opened the pull request.

majecty commented on test/src/e2e.dynval/dv.changeParams.test.ts:
```diff
@@ -0,0 +1,224 @@
+import { withNodes, changeParams, defaultParams } from "./setup";
+import { faucetSecret, faucetAddress } from "../helper/constants";
+
+async function getPossibleAuthors(
```
Please use the stake's getPossibleAuthors.

HoOngEe replied:
Changed

majecty commented on test/src/e2e.dynval/dv.changeParams.test.ts:
```diff
@@ -0,0 +1,224 @@
+            ))!;
+            const prevBlock1 = (await checkingNode.sdk.rpc.chain.getBlock(
+                termMetadata1.lastTermFinishedBlockNumber - 1
+            ))!;
```
Could you use a more descriptive variable name? It is hard to understand the prevBlock is whose prev block.

HoOngEe replied:
I changed the variable names.

HoOngEe commented:
I have implemented 3 test cases for the suite `changeParams`. I found an election error in the case that I changed the parameter `min_num_of_validators`. It was expected that 6 candidates would have been elected as validators but there were 7 validators in the next term.

foriequal0 commented on test/src/e2e.dynval/dv.changeParams.test.ts:
```diff
@@ -0,0 +1,224 @@
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
+                ...defaultParams,
+                ...overrideParams
```
You can write like this to reduce indirection.
```ts
const changeTxHash = await changeParams(checkingNode, 1, {
    ...defaultParams,
    termSeconds: 10,
```
And with this code, you could have overwritten some parameters. You should do instead:
```ts
const initialParams = {
    ...defualtParams,
    termSeconds: 20,
    maxNumOfValidators: 8
};
const node = withNodes(..., { overrideParams: initialParams });

it(..., async function() {
    const changeTxHash = await changeParams(checkingNode, 1, {
        ...initialParams,
        termSeconds: 10
    });
});
```

HoOngEe replied:
I applied it.

foriequal0 commented on test/src/e2e.dynval/dv.changeParams.test.ts:
```diff
@@ -0,0 +1,224 @@
+            const overrideParams = {
+                minPayCost: 11
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
```
ditto

foriequal0 commented on test/src/e2e.dynval/dv.changeParams.test.ts:
```diff
@@ -0,0 +1,224 @@
+                minNumOfValidators: 6,
+                termSeconds
+            };
+            const changeTxHash = await changeParams(checkingNode, 1, {
```
ditto

Remagpie commented on test/src/e2e.dynval/dv.changeParams.test.ts:
```diff
@@ -134,4 +135,71 @@ describe("Change commonParams", function() {
+        expect(validatorsAfter).contains.all.members(target);
+    }
+
+    describe.only("Change the minimum number of validators", async function() {
```
Remove `only`
